components/blitz/src/omero/gateway/facility/TablesFacility.java

```diff
@@ -252,6 +215,144 @@ public TableData getTable(SecurityContext ctx, long fileId, long rowFrom,
     }
     
     /**
+     * Perform a query on the table
+     * 
+     * @param ctx
+     *            The {@link SecurityContext}
+     * @param fileId
+     *            The id of the {@link OriginalFile} which stores the table
+     * @param condition
+     *            The query string
+     * @return A list of row indices matching the condition
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public long[] query(SecurityContext ctx, long fileId,
+            String condition) throws DSOutOfServiceException, DSAccessException {
+        return query(ctx, fileId, condition, 0, 0, 0);
+    }
+
+    /**
+     * Perform a query on the table
+     * 
+     * @param ctx
+     *            The {@link SecurityContext}
+     * @param fileId
+     *            The id of the {@link OriginalFile} which stores the table
+     * @param condition
+     *            The query string
+     * @param start
+     *            The index of the first row to consider (optional)
+     * @param stop
+     *            The index of the last+1 row to consider (optional)
+     * @param step
+     *            The stepping interval between the start and stop rows to
+     *            consider (Set to 0 to disable stepping)
+     * @return A list of row indices matching the condition
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public long[] query(SecurityContext ctx, long fileId,
+            String condition, long start, long stop, long step)
+            throws DSOutOfServiceException, DSAccessException {
+        TablePrx table = null;
+        try {
+            OriginalFile file = new OriginalFileI(fileId, false);
+            SharedResourcesPrx sr = gateway.getSharedResources(ctx);
```

Comment:
Doesn't have a `close()` method. Am I missing something, or does that also mean the example https://www.openmicroscopy.org/site/support/omero5.3/developers/Java.html is wrong?